numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -3927,7 +3927,7 @@ initialize_casting_tables(void)
     }
 
     _npy_can_cast_safely_table[NPY_STRING][NPY_UNICODE] = 1;
-    _npy_can_cast_safely_table[NPY_BOOL][NPY_TIMEDELTA] = 1;
+    _npy_can_cast_safely_table[NPY_BOOL][NPY_TIMEDELTA] = 0;
```

Comment:
I think this should go on line 3916 instead. At the same time, we should probably add a comment saying `except datetime + timedelta which are ambiguous without units`